mypy/typeanal.py

```diff
@@ -381,10 +396,40 @@ def analyze_callable_type(self, t: UnboundType) -> Type:
             ret_type = self.anal_type(t.args[1])
             if isinstance(t.args[0], TypeList):
                 # Callable[[ARG, ...], RET] (ordinary callable type)
-                args = t.args[0].items
+                args = []   # type: List[Type]
+                names = []  # type: List[str]
+                kinds = []  # type: List[int]
+                for arg in t.args[0].items:
+                    if isinstance(arg, CallableArgument):
+                        args.append(arg.typ)
+                        names.append(arg.name)
+                        if arg.constructor is None:
```

Comment:
We allow constructor to be None here; we can make it not be a fake string back in `fastparse.py`